snc.sh

```diff
@@ -266,16 +265,21 @@ function delete_operator() {
         ${OC} wait --for=delete pod/${pod} --timeout=120s -n ${namespace} || ${OC} delete pod/${pod} --grace-period=0 --force -n ${namespace} || true
 }
 
-if ! which ${UNZIP}; then
-    sudo yum -y install /usr/bin/unzip
+# All platforms use linux oc binary
+mkdir -p openshift-clients/linux
+curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux.tar.gz" | tar -zx -C openshift-clients/linux oc
+
+# Download the oc binary for mac and windows
+if [ "${ARCH}" != "ppc64le" ]; then
+	mkdir -p openshift-clients/mac openshift-clients/windows
+	curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-mac.tar.gz" | tar -zx -C openshift-clients/mac oc
+	curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-windows.zip" > openshift-clients/windows/oc.zip
+	if ! which unzip; then
+		sudo yum -y install /usr/bin/unzip
+	fi
+	unzip -o -d openshift-clients/windows/ openshift-clients/windows/oc.zip
 fi
 
-# Download the oc binary for all platforms
-mkdir -p openshift-clients/linux openshift-clients/mac openshift-clients/windows
-curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux.tar.gz" | tar -zx -C openshift-clients/linux oc
-curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-mac.tar.gz" | tar -zx -C openshift-clients/mac oc
-curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-windows.zip" > openshift-clients/windows/oc.zip
-${UNZIP} -o -d openshift-clients/windows/ openshift-clients/windows/oc.zip
 OC=./openshift-clients/linux/oc
```

Comment:
I'd keep that `OC` line grouped with the extraction of `openshift-client-linux.tar.gz`.
However, https://github.com/code-ready/snc/pull/263#pullrequestreview-514421267 still seems to be unaddressed? These mac/windows clients exist in `${MIRROR}/${OPENSHIFT_RELEASE_VERSION}` even for ppc64le/s390, and as your change currently stands, `create_disk.sh` will fail because it's expecting these windows/macos binaries to exist.
I agree it's suboptimal to download these binaries on ppc64le, but I also don't think it's breaking anything? So I'm reluctant to take a patch removing this, but breaking create_disk.sh.

Reply:
ok sorry I misunderstood the review comment and will address this change in my next PR. I am thinking about just adding a case statement into createdisk.sh for different archs. Similar to what was added in `run_preflight_checks()` in snc.sh so we can build the bundles for the specific archs as needed. I'm not sure yet, but something similar to...


```
ARCH=$(uname -m)
case $ARCH in
    x86_64)
        generate_hyperKit
        generate_hyperV;;
    ppc64le|s390x)
        generate_ppc64le;;
    *)	
        echo "The host arch is ${ARCH}. This is not supported by SNC!" && exit 1;;
esac
```

I'll take a closer look at createdisk.sh and submit a more appropriate PR :)

Reply:
You'll probably have a `generate_libvirt` both on x86_64 and ppc64le, but otherwise makes sense.